docs/development/quickstart.rst

```diff
@@ -0,0 +1,365 @@
+        scrolling down to "All downloads" -> "Tools for Visual Studio".
+        In the installer, select the "Desktop development with C++" Workloads.
+
+Alternative options include:
+
+- Install the necessary components on the command line using `vs_BuildTools.exe <https://learn.microsoft.com/en-us/visualstudio/install/use-command-line-parameters-to-install-visual-studio?source=recommendations&view=vs-2022>`_.
+- Use the `WSL <https://learn.microsoft.com/en-us/windows/wsl/install>`_.
+
+**MacOS**
+
+Install the Developer Tools using ``xcode-select --install``.
+
+Further details and related information can be found at
+https://devguide.python.org/setup/#macos.
+
+**Linux**
+
+For Linux-based installations, you won't have to install any additional components.
+
+.. _contributing.forking:
+
+Create a clone of astropy
+-------------------------
+
+If you have not done so already, you will need your own copy of ``astropy`` to
+build it and/or contribute to the source. Astropy is hosted in the `astropy GitHub repository <https://www.github.com/astropy/astropy>`_ and you need to make a clone.
+
+If you *only want to build astropy* but not contribute, you can run the following and then skip to the :ref:`next section<create-isolated-env>`::
+
+    git clone https://github.com/astropy/astropy.git
+    cd astropy
+
+If you plan to contribute a pull request, you will need a `free GitHub account
+<https://github.com/signup/free>`_.
+
+First, create a `GitHub Fork
+<https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo>`_ by going to the `astropy project page <https://github.com/astropy/astropy>`_
+and hitting the ``Fork`` button.
+
+Next, clone your GitHub fork to your machine:
+
+.. code-block:: shell
+
+    git clone https://github.com/YOUR-USER-NAME/astropy.git
+    cd astropy
+    git remote add upstream https://github.com/astropy/astropy.git
+    git fetch upstream
+
+This creates the directory ``astropy`` and connects your repository to the upstream
+(main project) `astropy <https://github.com/astropy/astropy>`__ repository.
+
+.. _create-isolated-env:
+
+Create an isolated development environment
+------------------------------------------
+
+A key requirement is to have an isolated Python environment, meaning that it is
+isolated from both your system Python and any other Python environments you may have
+for doing other work. This is important because the development environment will often
+be unstable and possibly broken at times, and you don't want to break your other work.
+
+There are many good options for doing this, including a number of virtual environment
+managers (e.g. the Python built-in `venv <https://docs.python.org/3/library/venv.html>`_
+module). Users who have a preference for a particular virtual environment manager are
+encouraged to use it!
+
+For this quickstart guide we use the `conda <https://docs.conda.io/en/latest/>`_ package
```

Comment:
#16555 is for installing a release version of astropy. This PR deals with installing astropy from source with the intent of developing code or docs, running tests etc. Overall, this PR is a huge change and if we have a long discussion about every small detail it will never converge.